craft_parts/plugins/cargo_package_plugin.py

```diff
@@ -0,0 +1,162 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2025 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""A plugin for packaging rust crates and publishing them to a local registry.."""
+
+import logging
+import pathlib
+import textwrap
+from typing import Literal, cast
+
+from overrides import override
+
+from craft_parts.constraints import UniqueList
+
+from . import validator
+from .base import Plugin
+from .properties import PluginProperties
+
+logger = logging.getLogger(__name__)
+
+CARGO_TOML_TEMPLATE = """
+[net]
+offline = true
+
+[source.craft-parts]
+directory = "{registry_dir}"
+
+[source.apt]
+directory = "/usr/share/cargo/registry"
+
+[source.crates-io]
+replace-with = "craft-parts"
+"""
+
+
+class CargoPackagePluginProperties(PluginProperties, frozen=True):
+    """The part properties used by the cargo-package plugin."""
+
+    plugin: Literal["cargo-package"] = "cargo-package"
+
+    # part properties required by the plugin
+    cargo_package_features: UniqueList[str] = []
```

Comment:
Maybe for the frozen class is not that important, but you may want to update it.

```suggestion
    cargo_package_features: UniqueList[str] = Field(default_factory=list)
```